backend/app/controllers/uoas.js

```diff
@@ -66,23 +67,52 @@ module.exports = {
         });
     },
 
-    insertOne: function (req, res, next) {
+    insert: function (req, res, next) {
         var thunkQuery = req.thunkQuery;
+        var uoas = req.body.name.split(/\s*,\s*/);
+        var sqlString = "'"+uoas.toString().replace(/'/g, "''").replace(/,/g, "','")+"'";
+
         co(function* () {
-            req.body.creatorId = req.user.realmUserId; // add from realmUserId instead of user id
-            req.body.ownerId = req.user.realmUserId; // add from realmUserId instead of user id
-            req.body.created = new Date();
-            return yield thunkQuery(UnitOfAnalysis.insert(req.body).returning(UnitOfAnalysis.id));
+            var added = yield thunkQuery(
+                'SELECT name, id FROM "UnitOfAnalysis" WHERE LOWER("UnitOfAnalysis"' +
+                '."name") IN (' + sqlString.toLowerCase() + ') AND "UnitOfAnalysis"' +
+                '."unitOfAnalysisType" = ' + req.body.unitOfAnalysisType
+            );
+
+            var insert = _.difference(uoas, added.map((exist) => exist.name));
+            for (var i = 0; i < insert.length; i++) {
+                var result = yield thunkQuery(UnitOfAnalysis.insert({
+                    name: insert[i],
+                    creatorId: req.user.realmUserId,
+                    ownerId: req.user.realmUserId,
+                    unitOfAnalysisType: req.body.unitOfAnalysisType,
```

Comment:
This fails is no `unitOfAnalysisType` is given. Would be a good idea to check that it exist first so we can send back the appropriate error message as opposed to getting back `"column \"undefined\" does not exist"`

Reply:
For the time being we're hardcoding Type to 1, which we know will exist. The check becomes more important after we start doing different types, but this works for now.